### PR TITLE
LLM - improved feature flags edition

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

Improved feature flags edition on LLM

https://github.com/LedgerHQ/ledger-live/assets/17146928/0edb5711-7979-447e-97fa-0482caff0cc3

### ❓ Context

- **JIRA or GitHub link**: [LIVE-11118]

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-11118]: https://ledgerhq.atlassian.net/browse/LIVE-11118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/FeatureFlagsSettings/FeatureFlagEdit.tsx [hunk 1]
```diff
@@ -28,88 +35,114 @@ const FeatureFlagEdit: React.FC<{
...
-  );
+  const [featureFlagValue, setFeatureFlagValue] = useState<Feature>(pureValue);
+  const [inputValueStringified, setInputValueStringified] = useState(formatValue(pureValue));
 
-  const inputValueDefaulted = inputValue || stringifiedPureValue;
-
-  const featureFlagsProvider = useFeatureFlags();
-
-  const { t } = useTranslation();
+  console.log(formatValue({ pureValue, featureFlagValue }));
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/screens/FeatureFlagsSettings/FeatureFlagEdit.tsx:41]: 🗑️

File: apps/ledger-live-mobile/src/screens/FeatureFlagsSettings/FeatureFlagEdit.tsx [hunk 2]
```diff
@@ -28,88 +35,114 @@ const FeatureFlagEdit: React.FC<{
...
 
-  const handleInputChange = useCallback((value: string) => {
-    setError(undefined);
-    setInputValue(value);
-  }, []);
+  const handleInputChange = (value: string) => {
+    try {
+      setInputValueStringified(value);
+      // Nb if value is invalid or missing, JSON parse will fail
+      JSON.parse(value);
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/screens/FeatureFlagsSettings/FeatureFlagEdit.tsx:54]: Never used after?

File: apps/ledger-live-mobile/src/screens/FeatureFlagsSettings/FeatureFlagEdit.tsx [hunk 3]
```diff
@@ -28,88 +35,114 @@ const FeatureFlagEdit: React.FC<{
...
+      setError("Bad Format");
+    }
+  };
 
   const handleRestoreFeature = useCallback(() => {
     setError(undefined);
-    setInputValue(undefined);
     featureFlagsProvider.resetFeature(flagName);
+    // setFeatureFlagValue(pureValue);
+    // setInputValueStringified(formatValue(pureValue));
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/screens/FeatureFlagsSettings/FeatureFlagEdit.tsx:65]: 🗑️

**Comment by mcayuelas-ledger** [comment]
Possible to provide screenshot?

**Comment by gre** [comment]
LLM codechecks issues

**Review by mcayuelas-ledger — APPROVED** [review]

**Pull request merged**